### PR TITLE
Explain that no struct can be created as transaction input other than some exceptions

### PR DESCRIPTION
**Pull request by nikos-terzo**
I happened to waste some time in my move learning process trying to see whether structs with `copy` ability could be created from the client side and passed into MoveVM. A colleague did the same thing too. Maybe we need to make this explicit.

I am not sure whether the format and places I used are the best.

### CODE REVIEW

File: book/src/concepts/what-is-a-transaction.md [hunk 1]
```diff
@@ -29,6 +29,31 @@ Transactions consist of:
...
+
+## Inputs
+
+Transaction inputs are the arguments for the transaction and are split between 2 types:
+- Pure arguments: These are mostly [primitive types](../move-basics/primitive-types.html) with some
+extra additions. A pure argument can be:
+    - boolean
+    - integer (u8, u16, u32, u64, u128, u256)
+    - address
+    - std::ascii::String
```
Comment (nikos-terzo) [inline book/src/concepts/what-is-a-transaction.md:44]: I am not really sure for this. @damirka , can you confirm?
Also have I left out any type that can be a pure argument?

File: book/src/concepts/what-is-a-transaction.md [hunk 2]
```diff
@@ -29,6 +29,27 @@ Transactions consist of:
...
+    - std::string::String
+    - vector of pure types
+    - Option of pure types
```
Comment (damirka) [inline book/src/concepts/what-is-a-transaction.md:43]: Keep full type definition as in `std::option::Option<T>` where `T` is a pure type.

File: book/src/concepts/what-is-a-transaction.md [hunk 3]
```diff
@@ -29,6 +29,27 @@ Transactions consist of:
...
+    - ID, a special type that refers to a unique identifier of an object
```
Comment (damirka) [inline book/src/concepts/what-is-a-transaction.md:44]: Ditto, not a special type, defined in `sui::object::ID`.

File: book/src/programmability/bcs.md [hunk 4]
```diff
@@ -52,6 +52,10 @@ Structs encode similarly to simple types. Here is how to encode a struct using B
 {{#include ../../../packages/samples/sources/programmability/bcs.move:encode_struct}}
 ```
 
+Note that encoding a custom struct does not make much sense in interacting with the Move VM, as
```
Comment (damirka) [inline book/src/programmability/bcs.md:55]: I'm not sure if this note is in place, there's no function available to developer which would deconstruct a struct from BCS.

Comment (nikos-terzo) [reply book/src/programmability/bcs.md:55]: There is not a single function but I just tested that something like this is indeed working:
```
    const coinDecoder = bcs.struct(
        '0x2::coin::Coin<0x2::sui::SUI>',
        {
            id: bcs.struct('0x2::object::UID', {
                id: bcs.struct('0x2::object::ID', {
                    bytes: bcs.Address
                }),
            }),
            balance: bcs.struct('0x2::balance::Balance<0x2::sui::SUI>', {
                value: bcs.u64()
            })
        }
    );

    for (const coin of coinsResp.data) {
        if (coin.data?.bcs?.dataType !== 'moveObject') {
            continue;
        }
        const data = fromB64(coin.data.bcs.bcsBytes);
        const coinData = coinDecoder.parse(data);
        console.log(coinData);
    }
```
What I am referring to here, is that while doing `bcs.struct` for the `parse` use-case has value, using it for `toBytes` use-case has none. Maybe this is not so clear currently?

Comment (damirka) [reply book/src/programmability/bcs.md:55]: Well, `toBytes` or its Move equivalent `bcs::to_bytes` has a number of applications. Main one - encoding transaction inputs / data for bridge-like applications. I suggest removing these edits from the scope of this PR, and focus on the corrections for the transaction inputs part.

File: book/src/concepts/what-is-a-transaction.md (hunk 2 above)
Comment (nikos-terzo) [reply book/src/concepts/what-is-a-transaction.md:43]: Should I also link the appropriate chapters on strings, address, vector, integers/booleans (primitive types)?

File: book/src/concepts/what-is-a-transaction.md (hunk 3 above)
Comment (nikos-terzo) [reply book/src/concepts/what-is-a-transaction.md:44]: object/object-model seems more appropriate as it refers to the UID.

File: book/src/programmability/bcs.md (hunk 4 above)
Comment (nikos-terzo) [reply book/src/programmability/bcs.md]: Resolved in [5faa5e2](https://github.com/MystenLabs/move-book/pull/91/commits/5faa5e2e5dd18255df69c4260ca8370373a427c7)

File: book/src/concepts/what-is-a-transaction.md (hunk 2 above)
Comment (nikos-terzo) [reply book/src/concepts/what-is-a-transaction.md]: [a49b910](https://github.com/MystenLabs/move-book/pull/91/commits/a49b910dc4a8537d8bacbc1b989f34b7c5004ab3)

File: book/src/concepts/what-is-a-transaction.md (hunk 3 above)
Comment (nikos-terzo) [reply book/src/concepts/what-is-a-transaction.md]: After all I linked `storage/uid-and-id`. I also referred to `object/object-model` as "See also".
[a49b910](https://github.com/MystenLabs/move-book/pull/91/commits/a49b910dc4a8537d8bacbc1b989f34b7c5004ab3)

File: book/src/concepts/what-is-a-transaction.md [hunk 5]
```diff
@@ -29,6 +29,27 @@ Transactions consist of:
...
+- Object arguments: These are objects or references of objects that the transaction will access. An
+object argument needs to be either a shared object, a frozen object, or an object that the
+transaction sender owns, in order for the transaction to be successfull.
+For more see [Object Model](http://localhost:3000/object/index.html).
+
+Note that a single command inside a transaction can also have a previous command's result as input.
```
[the remark this answers is not in the document]
Comment (nikos-terzo) [reply book/src/concepts/what-is-a-transaction.md]: Removed in [3118ca9](https://github.com/MystenLabs/move-book/pull/91/commits/3118ca92a8f4a4922b00c5379cbe2efa0477ac13). Just below in the commands section there exists the following phrase:
> The commands are executed in the order they are listed in the transaction, and they can use the results of the
previous commands, forming a chain

File: book/src/concepts/what-is-a-transaction.md [hunk 6]
```diff
@@ -29,6 +29,24 @@ Transactions consist of:
...
+- Pure arguments: These are mostly [primitive types](../move-basics/primitive-types.html) with some
+extra additions. A pure argument can be:
+    - [`bool`](../move-basics/primitive-types.html#booleans).
+    - [integer](../move-basics/primitive-types.html#integers) (`u8`, `u16`, `u32`, `u64`, `u128`, `u256`).
+    - [`address`](../move-basics/address.html).
+    - [`std::string::String`](../move-basics/string.html), UTF8 strings.
+    - [`std::ascii::String`](../move-basics/string.html#ascii-strings), ASCII strings.
+    - [`vector<T>`](../move-basics/vector.html), where `T` is a pure type.
+    - [`std::option::Option<T>`](../move-basics/option.html), where `T` is a pure type.
+    - [`std::object::ID`](../storage/uid-and-id.html), takes part in the Unique IDentifier (UID) of an object. See also [What is an Object](../object/object-model.html).
```
Comment (damirka) [inline book/src/concepts/what-is-a-transaction.md:44]: I wouldn't mention anything UID here. ID is just a type which typically points to an object, though it's not a requirement.

**Review by damirka — COMMENTED** [review]
Love the changes with small note on ID->UID

File: book/src/concepts/what-is-a-transaction.md (hunk 6 above)
Comment (nikos-terzo) [reply book/src/concepts/what-is-a-transaction.md]: [621aff8](https://github.com/MystenLabs/move-book/pull/91/commits/621aff82c8207ec169158b902bec6771bfa622a5)

**Review by damirka — APPROVED** [review]

**Comment by damirka** [comment]
@nikos-terzo LGTM! Thank you for doing this and for carrying out the edit process!

**Pull request merged**